### PR TITLE
Plugin order

### LINKED ISSUE #59: Reload JS only if included on the page
When the LiveReload client receives change notifications for CSS and images, it attempts to determine if those changes are relevant to the page and only reloads if they are. However, it doesn't do this for JS or other files, but rather [blindly reloads](https://github.com/livereload/livereload-js/blob/e1d943628005ad8d18a50ee1e8c29858ca748d10/dist/livereload.js#L777) the page.

This is problematic for build processes where an edit to a script that is included on the page may cause changes to other files that _aren't_ included on the page. (Perhaps you're editing a dependency of multiple build targets, or are generating sourcemaps alongside your scripts.)

I've developed a plugin that will intercept all changed JS and sourcemaps and only reload the page if the page includes a script tag with a matching filename: https://github.com/mixmaxhq/livereload-require-js-includes (So it
will never reload for a sourcemap change, the assumption there being that the
sourcemap would only have changed if the JS changed too.) Is this functionality that you'd consider bringing into the core project? It seems (perhaps naively?) like the way that LiveReload _should_ work.

The only downside I see is the difficulty of determining whether the JS is included on the page. My plugin currently only checks [the last path component](https://github.com/mixmaxhq/livereload-require-js-includes/blob/075f4941b4e111cd6633168c81a825fa91f3bb97/index.js#L18), but it appears like LiveReload includes a [sophisticated algorithm](https://github.com/livereload/livereload-js/blob/e1d943628005ad8d18a50ee1e8c29858ca748d10/dist/livereload.js#L719) for comparing paths and that could probably be employed here.

### PR DESCRIPTION
**Pull request by apla**
As of version 3.0.1 (and earlier) plugin order is always the same:

1. external plugins in order by `LiveReload.addPlugin()`;
2. liveCSS;
3. liveImg;
4. browser extension;
5. all other files

So, if no other plugin claims file ownership, page is always reloaded.

This patch allows to change plugin order and even turn off some plugins (like **all other files** _plugin_).

To activate plugin order, message should contain `pluginOrder` option with array of plugin identifiers. Those identifiers should resolve to internal _plugins_ (`css`, `img`, `extension` and `others`) or external ones (by `plugin.identifier`).

Fixes #59

### CODE REVIEW

File: src/livereload.js [hunk 1]
```diff
@@ -128,7 +128,8 @@ class LiveReload {
       reloadMissingCSS: message.reloadMissingCSS != null ? message.reloadMissingCSS : true,
       originalPath: message.originalPath || '',
       overrideURL: message.overrideURL || '',
-      serverURL: `http://${this.options.host}:${this.options.port}`
+      serverURL: `http://${this.options.host}:${this.options.port}`,
+      pluginOrder: message.pluginOrder
```
Comment (smhg) [inline src/livereload.js]: Possibly naive question: isn't `pluginOrder` something you'd want to provide in the script tag?
Would you change it with individual messages?

Comment (apla) [reply src/livereload.js]: Actually, I thinking about the same thing yesterday, this should be per-page setting, not per-update. I will change this behaviour.

File: src/livereload.js [hunk 2]
```diff
@@ -128,7 +128,8 @@ class LiveReload {
       reloadMissingCSS: message.reloadMissingCSS != null ? message.reloadMissingCSS : true,
       originalPath: message.originalPath || '',
       overrideURL: message.overrideURL || '',
-      serverURL: `http://${this.options.host}:${this.options.port}`
+      serverURL: `http://${this.options.host}:${this.options.port}`,
+      pluginOrder: this.options.pluginOrder
```
[the remark this answers is not in the document]
Comment (apla) [reply src/livereload.js]: Right! I completely forgot about this. Fixed in new commits. Also I've added short note in documentation about `pluginOrder`. To me, everything is covered.

**Comment by smhg** [comment]
Thanks! I've only now been able to run the tests over here. There seems to be some output that gets mixed up in the test output. You have that too?
Thinks like:
`LiveReload found 0 LINKed stylesheets, 0 @imported stylesheets
LiveReload will not reload path '/abc.css' because the stylesheet was not found on the page and 'options.reloadMissingCSS' was set to 'false'.`

**Comment by apla** [comment]
Yes, I have this messages too. In my pull request I'm using live css reloading to ensure plugins in order, but [live css reloader gives me this log message](https://github.com/livereload/livereload-js/blob/master/src/reloader.js#L242). I've silenced those messages.

**Comment by smhg** [comment]
Do I understand it right: there was no test that included CSS reloading before and your tests does? If so, that's awesome of course.

**Comment by smhg** [comment]
I don't understand how that influences this message appearing. I'll have a deeper look at things asap so we can proceed with this.

**Comment by smhg** [comment]
I don't know how I missed your updates to this PR. My apologies for this and thank you for the prompt changes!

I will release a new minor version with this soon.

**Pull request merged**